### PR TITLE
Avatar redesign

### PR DESCRIPTION
**Pull request by sureshcjc**
moved image from lambda resize to user service
created new endpoint for new lambda to trigger media resize

### CODE REVIEW

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        
+        try {
+            // Infer the image type.
+            Matcher matcher = Pattern.compile(".*\\.([^\\.]*)").matcher(sourceKey);
+            if (!matcher.matches()) {
+                logger.debug("Unable to infer image type for key {}", sourceKey);
+                return null;
+            }
+            String imageType = matcher.group(1);
+            if (!(JPG_TYPE.equals(imageType)) && !(PNG_TYPE.equals(imageType))) {
```
Comment (davidlaroue) [inline src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: Hmm what about JPEG?

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        } catch (Exception e) {
+            logger.error("Error processing image {}", e.getMessage());
+            throw new RuntimeException(e);
+        }
+        finally {
+            if(objectData!=null) { try {
+                objectData.close();
+            } catch (IOException e) {
+                logger.error("Error closing inputstream for s3 object");
+            } }
```
Comment (davidlaroue) [inline src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: looks like an extra bracket

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    private BufferedImage resizeImage(BufferedImage srcImage, int width, int height) throws IOException {
+        logger.debug("Source Image---->>>> {} ", srcImage);
+        int srcHeight = srcImage.getHeight();
+        int srcWidth = srcImage.getWidth();
+
+        float scalingFactor = Math.min(width / srcWidth, height / srcHeight);
+        int finalWidth = (int) (scalingFactor * srcWidth);
+        int finalHeight = (int) (scalingFactor * srcHeight);
+
+        BufferedImage resizedImage = new BufferedImage(finalWidth, finalHeight, BufferedImage.TYPE_INT_RGB);
```
[the remark this answers is not in the document]
Comment (davidlaroue) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: Hmm yeah I think we used imagscalr last time. I'm all for your current solution. It's just something we should remember to check if we notice anything weird with resized images. 

So should be okay to move forward with your current solution and if we see images that are all white lets revisit this.

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import com.victorious.user.model.ImageUrl;
+import com.victorious.user.service.UserMediaService;
+import com.victorious.user.service.UserService;
+
+@Service
+public class UserMediaServiceImpl implements UserMediaService {
+
+    private static final Logger logger = LoggerFactory.getLogger(UserMediaServiceImpl.class);
+    
+    //TODO: probably an existing enum we should use for this?
```
[the remark this answers is not in the document]
Comment (sureshcjc) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: What do you think about this image specific?
http://commons.apache.org/proper/commons-imaging/

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    @Override
+    public List<ImageUrl> processAvatarImage(String sourceBucket, String sourceKey) {
+        InputStream objectData = null;
+        
+        try {
+            // Infer the image type.
+            Matcher matcher = Pattern.compile(".*\\.([^\\.]*)").matcher(sourceKey);
+            if (!matcher.matches()) {
+                logger.debug("Unable to infer image type for key {}", sourceKey);
+                return null;
```
[the remark this answers is not in the document]
Comment (sureshcjc) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: Agreed! Will get that in.

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java (hunk 2 above)
Comment (sureshcjc) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: looks good to me and no compile errors.

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (sureshcjc) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: FYI, this one will be running async post upload. user upload is returned after s3 bucket save which triggers the labmda to call this service to resize and update user avatar urls.

File: src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java (hunk 2 above)
Comment (davidlaroue) [reply src/main/java/com/victorious/user/service/impl/UserMediaServiceImpl.java]: Aah yeah couldn't see try was on same line as if in the pr. 👍

**Pull request merged**